src/core/dbus-unit.c

```diff
@@ -1867,6 +1867,56 @@ static int bus_unit_set_live_property(
                 return 1;
         }
 
+        /* A settings that only apply to active units. We don't actually write this to /run, this state is
+         * managed internally. "+foo" sets flag foo, "-foo" unsets flag foo, just "foo" resets flags to
+         * foo. The last type cannot be mixed with "+" or "-". */
+
+        if (streq(name, "Markers")) {
+                _cleanup_free_ char **s = NULL;
+                char **word;
+                UnitMarker settings = 0, mask = 0;
+                bool some_plus_minus = false, some_absolute = false;
+
+                r = sd_bus_message_read_strv(message, &s);
```

Comment:
doesn't matter, but instead of reading the whole thing into an strv and then iterating through it and freeing it again i'd probably just iterate through it one by one right. the code is not that much longer, but should be more efficeint:

```c
r = sd_bus_message_enter(message, 'a', "s");
if (r < 0)
        return r;

for (;;) {
        const char *word;
        r = sd_bus_message_read(message, "s", &word);
        if (r < 0) 
                return r;
        …
}
r = sd_bus_message_exit_container(m);
if (r < 0)
        return r;
```